### PR TITLE
Add Github Action required files and test workflow

### PR DESCRIPTION
**Pull request by yongyan-gh**
per #49

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    steps:
+      # To use this repository's private action,
+      # you must check out the repository
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Flawfinder action step
+        uses: ./ # Uses an action in the root directory
+        with:
+          command: '--version'
```
Comment (eddynaka) [inline .github/workflows/main.yml:16]: >--version [](http://example.com/codeflow?start=20&length=9)

(1) change to the command to analyze and export as sarif
(2) add the  step to send to github

**Comment by eddynaka** [comment]
@yongyan-gh , thanks for creating the pr!

@david-a-wheeler , this will give users the ability to run flawfinder and export the way they want.
Once it gets merged and published, we will create a second PR in https://github.com/actions/starter-workflows, adding the complete example using the analyze command, exporting as sarif, and uploading to github.

**Comment by david-a-wheeler** [comment]
Overall great stuff!

Is there an easy way to let users control how flawfinder is executed for their particular project? Maybe some settings can be brought in through an environment variable or some such? At the least, can the user replace the arguments provided without rewriting the GitHub Action?

Also, since this is for a CI/CD pipeline, I'm guessing users want `--error-level=4` (perhaps replace 4 with 5). That way the pipeline will fail by default with a high-level vulnerabilities. Again, this should configurable without creating a different GitHub action.

**Comment by david-a-wheeler** [comment]
Also: Can you briefly document "how to use it"? If nothing else, write some text & I'll figure out how to format it. But I want to tell users at least briefly how to enable it & control it.

**Comment by eddynaka** [comment]
> Overall great stuff!
> 
> Is there an easy way to let users control how flawfinder is executed for their particular project? Maybe some settings can be brought in through an environment variable or some such? At the least, can the user replace the arguments provided without rewriting the GitHub Action?
> 
> Also, since this is for a CI/CD pipeline, I'm guessing users want `--error-level=4` (perhaps replace 4 with 5). That way the pipeline will fail by default with a high-level vulnerabilities. Again, this should configurable without creating a different GitHub action.

Hi @david-a-wheeler , the idea of the action is to give the user the possibility to use the way they want.

So, for example, setup-dotnet action you can pass the version you want to use.

For this, you can pass errors only or other things.

The idea is:
1. we would completely separate flawfinder from github, because it can enable the user to just upload as artifacts and as csv 
2. create a full workflow in the starter actions, that will give the boilerplate workflow that scans and push to github

**Comment by eddynaka** [comment]
> Also: Can you briefly document "how to use it"? If nothing else, write some text & I'll figure out how to format it. But I want to tell users at least briefly how to enable it & control it.

@yongyan-gh , can you update the readme?

we can add a section to it saying how to use it.

**Comment by david-a-wheeler** [comment]
I need to release a new version to bring in #52. I'm thinking about merging this pull request (#51), then releasing, so that we have at least what's done so far. I do still want some better docs eventually, but if we're going to cut a release anyway, it seems reasonable to have this first cut included. Comments?

**Comment by eddynaka** [comment]
@david-a-wheeler , let's wait @yongyan-gh update the PR adding a section showing how to use the action in the readme.
once he does that and we merge #52, everything will flow naturally.

**Comment by david-a-wheeler** [comment]
Okay, my plan is to release a new version tonight with #52 .

**Comment by eddynaka** [comment]
@david-a-wheeler , we added the section to explain how to use the action.
Pls, take a look :)

File: README.md [hunk 2]
```diff
@@ -94,6 +94,27 @@ vulnerabilities in programs that cannot be built or cannot be linked.
...
+# Flawfinder Github Action
+
+## Usage
+
+See [action.yml](https://github.com/david-a-wheeler/flawfinder/blob/main/action.yml)
+
+Create a .yml file under .github/workflows with the following contents:
+
+### Basic:
+````yml
```
Comment (eddynaka) [inline README.md:106]: remove one of the ` (just three is needed)

File: README.md [hunk 3]
```diff
@@ -94,6 +94,27 @@ vulnerabilities in programs that cannot be built or cannot be linked.
...
+- name: Flawfinder
+  uses: david-a-wheeler/flawfinder@v1.0
+  with:
+    arguments: '--sarif ./'
+    output: 'flawfinder_results.sarif'
+````
```
Comment (eddynaka) [inline README.md:112]: add a new line and remove the forth `

File: README.md [hunk 4]
```diff
@@ -94,6 +94,27 @@ vulnerabilities in programs that cannot be built or cannot be linked.
...
+### Input options:
+- arguments: Flawfinder command arguments. 
+Visit https://github.com/david-a-wheeler/flawfinder/blob/master/README.md#usage to check all parameters.
```
Comment (eddynaka) [inline README.md:115]: review the link, i think it will point to the wrong sectionn

**Comment by david-a-wheeler** [comment]
I have uploaded flawfinder 2.0.18, which I believe fixes the SARIF output issue. Hopefully that will make this pull request easier to test.

@yongyan-gh :  @eddynaka has provided a number of review comments. Please look them up over, fix (or reject if you don't agree), and let me know when you've done that. I definitely intend to merge this, but we all want it to be a good initial version.

THANK YOU everyone very much for your time!

File: .github/workflows/main.yml (hunk 1 above)
Comment (yongyan-gh) [reply .github/workflows/main.yml]: done

File: README.md (hunk 2 above)
Comment (yongyan-gh) [reply README.md]: updated

File: README.md (hunk 4 above)
Comment (yongyan-gh) [reply README.md:116]: I only found some documentation about argument in READ.ME usage section. @david-a-wheeler is there other link documents all Flawfinder arguments/parameters?

**Comment by yongyan-gh** [comment]
> I have uploaded flawfinder 2.0.18, which I believe fixes the SARIF output issue. Hopefully that will make this pull request easier to test.
> 
> @yongyan-gh : @eddynaka has provided a number of review comments. Please look them up over, fix (or reject if you don't agree), and let me know when you've done that. I definitely intend to merge this, but we all want it to be a good initial version.
> 
> THANK YOU everyone very much for your time!

Thanks for publishing the new release. Now I can see the workflow calls the action passed without the SARIF validation error.
This is one example in my test branch:
https://github.com/yongyan-gh/flawfinder/actions/runs/969842889

**Comment by david-a-wheeler** [comment]
@yongyan-gh : Great! @eddynaka has provided a number of review comments. You don't need to agree with them all, but I'd like to hear your responses (& changes where appropriate) before I merge. We're close, I want to make sure this gets merged AND make sure we get a good version merged.

**Comment by eddynaka** [comment]
> @yongyan-gh : Great! @eddynaka has provided a number of review comments. You don't need to agree with them all, but I'd like to hear your responses (& changes where appropriate) before I merge. We're close, I want to make sure this gets merged AND make sure we get a good version merged.

Hi @david-a-wheeler, I think @yongyan-gh already fixed the issues that I pointed out.

**Review by eddynaka — APPROVED** [review]

**Comment by david-a-wheeler** [comment]
Great! Merging!

**Pull request merged**